python/python/lance/torch/distance.py

```diff
@@ -214,6 +216,7 @@ def l2_distance(
             return _l2_distance(vectors, centroids, split_size=split)
         except RuntimeError as e:  # noqa: PERF203
             if "CUDA out of memory" in str(e):
+                logging.warning("L2: batch split=%s out of memory", split)
```

Comment:
since we're taking action automatically, can you add to the warning to let the user know that
1. We've automatically recovered from this
2. The action was that we reduced the batch size by half

Reply:
done